src/dotnet/Core/Blobs/Internal/GoogleCloudBlobStorage.cs

```diff
@@ -0,0 +1,97 @@
+using System.Net;
+using Google;
+using Google.Cloud.Storage.V1;
+
+namespace ActualChat.Blobs.Internal;
+
+internal class GoogleCloudBlobStorage : IBlobStorage
+{
+    private readonly string _bucket;
+    private readonly StorageClient _client;
+
+    public GoogleCloudBlobStorage(string bucket)
+    {
+        _bucket = bucket;
+        _client = StorageClient.Create();
+    }
+
+    public async Task<Stream?> Read(string path, CancellationToken cancellationToken)
+    {
+        if (path == null) throw new ArgumentNullException(nameof(path));
+
+        var stream = new MemoryStream();
```

Comment:
Can we also use here recycleable memory streams? Or does they not suit for this task?

Reply:
Let me check